libjulia/interpreter/EVMInstructionInterpreter.cpp

```diff
@@ -0,0 +1,485 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * IULIA interpreter module that evaluates EVM instructions.
+ */
+
+#include <libjulia/interpreter/EVMInstructionInterpreter.h>
+
+#include <libjulia/interpreter/Interpreter.h>
+
+#include <libsolidity/inlineasm/AsmData.h>
+
+#include <libevmasm/Instruction.h>
+
+#include <libdevcore/SHA3.h>
+#include <libdevcore/CommonData.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::julia;
+
+namespace
+{
+
+u256 exp256(u256 _base, u256 _exponent)
+{
+	using boost::multiprecision::limb_type;
+	u256 result = 1;
+	while (_exponent)
+	{
+		if (static_cast<limb_type>(_exponent) & 1)	// If exponent is odd.
+			result *= _base;
+		_base *= _base;
+		_exponent >>= 1;
+	}
+	return result;
+}
+
+u256 readZeroExtended(bytes const& _data, u256 const& _offset)
+{
+	if (_offset >= _data.size())
+		return 0;
+	else if (_offset + 32 <= _data.size())
+		return *reinterpret_cast<h256 const*>(_data.data() + size_t(_offset));
+	else
+	{
+		size_t off = size_t(_offset);
+		u256 val;
+		for (size_t i = 0; i < 32; ++i)
+		{
+			val <<= 8;
```

Comment:
Is this subject to the "boost shift bug"?

Reply:
I hope this is gone now...